EMongoModel.php

```diff
@@ -404,10 +404,15 @@ public function getRelated($name, $refresh = false, $params = array())
 				$result = array();
 				foreach($pk as $singleReference){
 					$row = $this->populateReference($singleReference, $cname);
-					if($row){
-						array_push($result, $row);
-					}
+					
+					// When $row does not exists it will return null. It will not add it to $result

```

Comment:
There are essentially two bugs:
1. That the one-relation returns an empty array when a document does not exists
2. When a record does not exists and is retrieved by MongoDbRef::get you get an error of the foreach in `populateRecord` via `populateReference`